### ISSUE TITLE
Improve one on one chat

### ISSUE THREAD

**Issue by rullzer** [nextcloud/talk-android#249]
Talk:

![screenshot_nextcloud_talk_20180719-184817](https://user-images.githubusercontent.com/45821/42958024-a564b7d0-8b84-11e8-8d2f-7363f094f7c5.png)

Signal (but other apps look similar)
![screenshot_signal_20180719-184033](https://user-images.githubusercontent.com/45821/42957794-f6cb940a-8b83-11e8-8e97-a8d644592943.png)

Main things I notice:

1. The avatar next to the name in the top bar
2. There is no need to show the name of the participant and their avatar on each message from them. This just eats up space and doesn't add new information
3. We should probably use fuzzy logic for the times as well (just now, a minute ago etc). At least for recent stuff the exact  time is not that important I think

@nextcloud/designers

**Comment by mario** [comment]
I'm working on some changes to the design so I'll poke you once that is ready.

**Comment by jancborchardt** [comment]
@mario if you are working on design changes, please involve @nextcloud/designers ;) Let us know what you are working on, it’s an iterative process.

**Comment by mario** [comment]
@jancborchardt I cannot do mockups clearly (at least not YET), so I'll do the changes and I'll then involve @nextcloud/designers.

Also, some of the changes were already discussed in another issue.

**Comment by jancborchardt** [comment]
> I cannot do mockups clearly (at least not YET), so I'll do the changes and I'll then involve 

You don’t need to do mockups, no worry. ;) But at least describing what you plan to do would be great, before you spend time on working something. Otherwise a lot of time might be wasted, and that’s not nice for anyone.

----

Also, if you _do_ want to do mockups, the recommended free software is [Pencil](https://pencil.evolus.vn/) or [UXBOX](https://www.uxbox.io/).

**Comment by mario** [comment]
@rullzer

1) IHMO the avatar in the nav bar pollutes spaces - instead whitespace + two icons already there completely visible is what I strongly prefer. (voice and video call).

2) #158 is implemented, participant name and avatar will be only be shown on the first message in a group. (e.g. group is a set of messages by the same participant in the same minute.

3) See 2. point on why I think 3. is redundant

**Comment by jancborchardt** [comment]
> 1) IHMO the avatar in the nav bar pollutes spaces - instead whitespace + two icons already there completely visible is what I strongly prefer. (voice and video call).

This is what people are used to from apps such as WhatsApp and Signal.

- In 1on1 conversations, the avatar from the other person is shown in the header next to the name, and there is no need to repeat it for every message.
- In a group conversation, it’s the group image and name up there, and then of course it shows the avatars and names for every message (or once for a set of subsequent messages).

This is not about our individual preferences or opinions, because we are designing and developing an app for thousands (and/or hopefully millions) of users. It’s already difficult enough being open source and needing setup, so our design needs to be at least on par with the other widely used proprietary and other open source apps.


> 2) Once #158 is implemented, participant name and avatar will be only be shown on the first message in a group. (e.g. group is a set of messages by the same participant in the same minute.

This is good of course. But for 1on1 chats, it still does not make sense to show the other person’s avatar every set of messages. That’s what the header is for, as the name already shows there – and next to the name belongs the person’s face.

**Comment by jancborchardt** [comment]
Another point why the image should be in the top bar: Viewing details of the conversation and editing them should happen through tapping the title bar (avatar & name part) and there you should also be able to edit both the conversation name _and_ a future group picture.

So showing the image there gives you the affordance or hint that this is a place for editing the picture (in the case of group) or viewing more of that person (in the case of 1on1).

**Comment by jancborchardt** [comment]
> Tapping the title bar to open "settings" is not a very Android-like action despite a few apps implementing this pattern.

Sorry but basically everyone in the messenger space is doing it and people are used to it. Sure we can add another action "Conversation settings" in the 3-dot-menu, but tapping the title should certainly also do that as a quick action. :)

**Comment by mario** [comment]
"Everyone does it" is not the real reason to do something.

**Comment by mario** [comment]
(P.S. I'll make you a deal - if you find the pattern you want in ANY of the official material design docs, I'd be more than happy to do it)

In addition, to stay on-topic for the avatar in nav bar, even the iOS guidelines say against this:

"Avoid crowding a navigation bar with too many controls. In general, a navigation bar should contain no more than the view’s current title, a back button, and one control that manages the view’s contents. If you use a segmented control in the navigation bar, the bar shouldn’t include a title or any controls other than the segmented control."

**Comment by jancborchardt** [comment]
@mario from https://material.io/design/components/app-bars-top.html#anatomy (section "Images in bars")

> **Images in bars**
> Bars can contain imagery. Prominent top app bars are recommended for image use because they provide more space.

The example they show is this:
![](https://storage.googleapis.com/spec-host-backup/mio-design%2Fassets%2F1UmsSQiiLgULeUnPsyvwdtixhQ7VfqRmN%2Ftopappbars-images.png)

So what we could do is show an enlarged (and blurred) image of the person in the header. Not with the goal of showing their face of course, but to just give a general feel to the conversation.

**Comment by mario** [comment]
I could do this. But keep in mind when you scroll this 'header' is hidden
and is not really meant for chat. But I will figure something out so
everyone is happy.

**Comment by jancborchardt** [comment]
> But I will figure something out so everyone is happy.

It would be really awesome if you simply take input from the designers, like the above https://github.com/nextcloud/talk-android/issues/249#issuecomment-406399409. :) It’s a team effort.

**Comment by mario** [comment]
Thank you for your input, like I mentioned above it has been taken into consideration, however my input has to be respected since I'm working on the product and I'm more aware of the overall Android UX.

**Comment by nickvergessen** [comment]
> 1. The avatar next to the name in the top bar
> 2. There is no need to show the name of the participant and their avatar on each message from them. This just eats up space and doesn't add new information

I checked all 6 Messengers I use, in group conversations they all are the same (like ours, apart from the timestamp being next to the bubble instead of inside, but i think its okay)
as for 1-1 chats, all show the name and avatar in the header, 3 show the avatar additionally on all messages, but no name.
So I would go with the masses here and add it to the header and remove the name + avatar from the messages.

> 3. We should probably use fuzzy logic for the times as well (just now, a minute ago etc). At least for recent stuff the exact time is not that important I think

I'd not do that at all, only signal does this, all others show the normal timestamp.

**Comment by mario** [comment]
@nickvergessen it will be done like that.
